lib/manageiq/appliance_console/message_configuration.rb

```diff
@@ -0,0 +1,134 @@
+
+        @keystore_dir_path                 = @config_dir_path.join("", "keystore")
+        @truststore_path                   = @keystore_dir_path.join("", "truststore.jks")
+        @keystore_path                     = @keystore_dir_path.join("", "keystore.jks")
+
+        @messaging_yaml_sample_path        = @miq_config_dir_path.join("", "messaging.kafka.yml")
+        @messaging_yaml_path               = @miq_config_dir_path.join("", "messaging.yml")
+      end
+
+      def create_tools_log_config
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        FileUtils.cp(@tools_log4_properties_sample_path, @tools_log4_properties_path) unless file_found?(@tools_log4_properties_path)
+      end
+
+      def create_client_properties
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        content = <<~CLIENT_PROPERTIES
+          ssl.truststore.location=#{@truststore_path}
+          ssl.truststore.password=#{@password}
+
+          sasl.mechanism=PLAIN
+          security.protocol=SASL_SSL
+          sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required \\
+            username=#{@username} \\
+            password=#{@password} ;
+        CLIENT_PROPERTIES
+
+        File.write(@client_properties_path, content) unless file_found?(@client_properties_path)
+      end
+
+      def configure_messaging_yaml
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        return true if file_found?(@messaging_yaml_path)
+
+        messaging_yaml = YAML.load_file(@messaging_yaml_sample_path)
+
+        messaging_yaml["production"]["hostname"] = @server_hostname
+        messaging_yaml["production"]["port"] = 9093
+        messaging_yaml["production"]["username"] = @username
+        messaging_yaml["production"]["password"] = @password
+
+        File.write(@messaging_yaml_path, messaging_yaml.to_yaml) unless file_found?(@messaging_yaml_path)
+      end
+
+      def remove_installed_files
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        @installed_files.each { |f| FileUtils.rm_rf(f) if File.exist?(f) }
+      end
+
+      def valid_environment?
+        return false unless installation_valid?
+
+        if already_configured?
+          return false unless agree("\nAlready configured on this Appliance, Un-Configure first? (Y/N): ")
+
+          deactivate
+          return false unless agree("\nProceed with Configuration? (Y/N): ")
+        end
+        true
+      end
+
+      def file_found?(path)
+        return false unless File.exist?(path)
+
+        say("\tWARNING: #{path} already exists. Taking no action.")
+        true
+      end
+
+      def file_contains?(path, content)
+        return false unless File.exist?(path)
+
+        content.split("\n").each do |l|
+          l.gsub!("/", "\\/")
+          l.gsub!(/password=.*$/, "password=") # Remove the passord as it can have special characters that grep can not match.
```

Comment:
Typo in comment - passord -> password

Reply:
@Fryguy Good catch! I ran spell checker. I wonder why it didn't flag this.